core/utils/json/json.go

```diff
@@ -59,6 +59,10 @@ func convertJsonNumbers(val any) any {
 			return val.String()
 		}
 		return i
+	case string:
+		return val
+	case int64:
+		return val
```

Comment:
Worth mentioning that these are just optimizations.  These types would end up in default and go through reflection to check sliceiness  only to return as-is.